### ISSUE TITLE
In-game note taking

### ISSUE THREAD

**Issue by drbig** [CleverRaven/Cataclysm-DDA#13541]
(_Full immersion FTW!_)

Items:
- Pen - has charges, scaling tbd
- Notepads - various size, random number of clean pages

Mechanic:
- You can write line-wise on clean pages of your notepad
- You can't edit, but you can cross-out stuff
- Lines have some fixed width, but number of lines may depend on INT (scaling tbd)
- You obviously need to be literate for all of the above

Code:
- Browser+editor as a simple UI with per-page view and lines addressed by a shortcut
- Probably save as another JSON, array of pages each with array of lines of content + crossed-out flag
- Need to tie it to particular item ID - is that doable?

The pen-only thing is for balance - no reusable notepads. It _is_ another anachronistic thing (we have PDAs)... but then if we were to get this in it'd be little work to actually make PDAs more useful :)
##

**Comment by stk2008** [comment]
Wow this is cool

**Comment by Coolthulhu** [comment]
> Lines have some fixed width, but number of lines may depend on INT (scaling tbd)

Scaling sounds like way more problem than it could be worth. And it isn't exactly realistic (intelligence doesn't help with handwriting).

> The pen-only thing is for balance - no reusable notepads.

Stuff that's 100% flavor doesn't really need any sort of balance.

**Comment by drbig** [comment]
> Scaling sounds like way more problem than it could be worth. And it isn't exactly realistic (intelligence doesn't help with handwriting).

Do agree.

> Stuff that's 100% flavor doesn't really need any sort of balance.

Mhm, it makes sense for pens (not going to even suggest implementing an eraser). But for PDAs we'd need editing anyway.

**Comment by chaosvolt** [comment]
Amusing, but not sure if this should be mandatory.

What next, needing an in-game map and compass? X3

**Comment by drbig** [comment]
@chaosvolt mandatory? Definitely not. It's a flavour thing.

**Comment by chaosvolt** [comment]
Ah, I see. So this would be supplementing map notes, not replacing them?

**Comment by drbig** [comment]
Exactly. Map notes are more like putting a circle and some super-short note. This would enable you to keep in-game stuff written down in-game. Mostly for role-play (keep a dairy?) - also it'd be super cool if we were to play succession games (you'd have previous player's notes right in game on your character :D).

**Comment by stk2008** [comment]
How about books

We can write whole books and place them on a book shelf.
Just like in mine craft.

Writing books would be epic

**Comment by FireCrow1973** [comment]
Wold be cool if you found a journey diary on survivor zombies, detailing a slow dissent in to madness.
Tons of creative writing opportunities, and a fun way to explore lore.

**Comment by chaosvolt** [comment]
Or something like more involved survivor notes.

**Comment by i2amroy** [comment]
>(keep a dairy?)

Mmmm, a new source of fresh milk! :P

That said definitely liking the idea.

**Comment by stk2008** [comment]
What would be even cooler is

Finding a note on a dead body and it gives you hints to a hidden stash
(Stalker rip off LOL).

**Comment by chaosvolt** [comment]
If this makes the stash generate only AFTER finding the note, I will be saddragon. XP

**Comment by stk2008** [comment]
LOL yeah very true I hate that in stalker its nearly as bad as enemys spawn
ontop of you from no where lol

**Comment by chaosvolt** [comment]
Or when they develop a case of Titanium Forehead Syndrome?

**Comment by stk2008** [comment]
Yeah a lot of mods fixed that though but I know what ya saying LOL.

**Comment by kevingranade** [comment]
Implementing the text storage would be fairly straightforward, mapping
functions to items is also pretty easy, also we could use the same
mechanism for pre-existing items with large amounts of text the player
would want to read.
As for the text editor itself... this is why we don't have this feature
already, I didn't feel like writing a text editor :)
Please do strive for utf8 cleanliness, we want everyone to be able to play
:)

As for maps and compasses in-game, it's not out of the question,
orienteering is a big part of survival :)

**Comment by stk2008** [comment]
Damn was just thinking I need to make notes and then thought oh yeah I see this a while ago.

I assume its not been PR`d :(

**Comment by Soyweiser** [comment]
Just fyi, this idea has been suggested before: #6197

**Comment by mugling** [comment]
feeping creaturism

**Comment by illi-kun** [comment]
### This issue was closed as it appears inactive.

Reducing open issues to those which are (or will) be actively worked upon helps us focus our efforts. This issue has not been deleted - it still appears in searches and if it contains relevant information you are encouraged to continue to link to it.

### If this issue was a bug
It should be reopened if it can be reproduced in the current build. You can obtain the most recent copy [here](http://dev.narc.ro/cataclysm/jenkins-latest/). Please check there is not a more recent report of this bug before doing so. If no more recent report exists you should continue the discussion in this issue.

### If this was a feature request
If the consensus was that the idea was good you could consider submitting an implementation via a PR. If you want to comment further please do so here as opposed to opening a new issue. Before posting check nobody has already made the same point and consider whether your comments are likely to lead to an implementation. If you have doubts about either consider instead [voting](http://iv.cataclysmdda.com:8111/all) for the issue

### If you want to work on this issue
Then either assign it to yourself or if you are unable to do so claim it via adding a comment. Please don't assign others or make a general request for action.
